### ISSUE TITLE
Django admin: make multi-selects comply to accessibility criteria

### ISSUE THREAD

**Issue by vickyszuchin** [cisagov/manage.get.gov#1949]
### Issue description

Make multi-selects comply to accessibility criteria. There is an accessibility issue on multi-selects (h2 should be an h3, or a <label> maybe)

See:
-  [slack thread](https://cisa-corp.slack.com/archives/C05BGB4L5NF/p1711644809098279) for screenshot
-  [slack thread](https://cisa-corp.slack.com/archives/C06K96473FA/p1711572361126209) for additional discussion

### Acceptance criteria

- [ ] ensure that multi-selects use h3 instead of h2, to be compliant with accessibility requirements.

### Additional context

Multi-selects are still used on User Model/Permissions.

Any accessibility audit that sees the structure:
```
<h2>Permissions</h2>
<h2>Available groups</h2>
<h2>Chosen groups</h2>
```
...
Will fail the page. It is structurally incorrect that Available groups and Chosen groups are sections on the level of Permissions on the User change_form 

### Links to other issues
Related to: #1950
